src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1582,12 +1586,59 @@ private void GatherArrayStringAndValueOutputs(bool outputTargetIsItem, string ou
                             }
                         }
 
+                        LogPropertyInTaskAssignment(outputTargetName, outputString, parameterLocation);
                         _batchBucket.Lookup.SetProperty(ProjectPropertyInstance.Create(outputTargetName, outputString, parameterLocation, _projectInstance.IsImmutable));
                     }
                 }
             }
         }
 
+        /// <summary>
+        /// Logs property assignment information during task execution, based on configured property tracking settings.
+        /// </summary>
+        /// <param name="propertyName">The name of the property being assigned or reassigned.</param>
+        /// <param name="propertyValue">The new value being assigned to the property.</param>
+        /// <param name="location">The source location where the property assignment occurs.</param>
+        private void LogPropertyInTaskAssignment(string propertyName, string propertyValue, IElementLocation location)
+        {
+            if (_propertyTrackingSettings == 0)
```

Comment:
Nit: compare against enum values for readability here:

```suggestion
            if (_propertyTrackingSettings == PropertyTrackingSetting.None)
```